source/gfx.h

```diff
@@ -148,24 +151,63 @@ extern uint8_t mul_brightness [16][32];
 
 static INLINE uint16_t COLOR_ADD(uint16_t C1, uint16_t C2)
 {
+#if defined(PSP)
+   /* PSP uses pre-1.60 colour operations */
    if (C1 == 0)
       return C2;
    else if (C2 == 0)
       return C1;
    else
       return GFX.X2[(((C1 & RGB_REMOVE_LOW_BITS_MASK) + (C2 & RGB_REMOVE_LOW_BITS_MASK)) >> 1) + (C1 & C2 & RGB_LOW_BITS_MASK)] | ((C1 ^ C2) & RGB_LOW_BITS_MASK);
+#else
+	const int RED_MASK   = 0x1F << RED_SHIFT_BITS;
+	const int GREEN_MASK = 0x1F << GREEN_SHIFT_BITS;
+	const int BLUE_MASK  = 0x1F;
+
+	int rb          = (C1 & (RED_MASK | BLUE_MASK)) + (C2 & (RED_MASK | BLUE_MASK));
+	int rbcarry     = rb & ((0x20 << RED_SHIFT_BITS) | (0x20 << 0));
+	int g           = (C1 & (GREEN_MASK)) + (C2 & (GREEN_MASK));
+	int rgbsaturate = (((g & (0x20 << GREEN_SHIFT_BITS)) | rbcarry) >> 5) * 0x1f;
+	uint16_t retval = (rb & (RED_MASK | BLUE_MASK)) | (g & GREEN_MASK) | rgbsaturate;
+
+#if GREEN_SHIFT_BITS == 6
+	retval         |= (retval & 0x0400) >> 5;
+#endif
+
+	return retval;
+#endif
```

Comment:
Just FYI, there is also a `COLOR_ADD_BRIGHTNESS` is Snes9x 1.60: https://github.com/snes9xgit/snes9x/blob/668f1d967bfa748f0a629e956226133a6f3bfa30/gfx.h#L158-L175

I don't think it is used in those older versions of Snes9x though.

Reply:
Thanks - but yes, this isn't used at all in the core :)